File: find-greatest-common-divisor-of-array.ts
{
    function findGCD(nums: number[]): number {
        const max = Math.max(...nums)
        const min = Math.min(...nums)
        
        for(let i = min; i > 0; i--) {
            if(max % i === 0 && min % i === 0) {
                return i
            }
        }
    };

    console.log(findGCD([2,5,6,9,10]))
}

File: find-greatest-common-divisor-of-array.rs
pub fn find_gcd(nums: Vec<i32>) -> i32 {
    let (mut min, mut max) = nums
        .into_iter()
        .fold((i32::MAX, i32::MIN), |(min, max), num| {
            (min.min(num), max.max(num))
        });

    while min.is_positive() {
        min = std::mem::replace(&mut max, min) % min;
    }
    max
}
